modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/js/PersonAccountEntryEventHandler.es.js

```diff
@@ -53,7 +53,7 @@ class PersonAccountEntryEventHandler extends PortletBase {
 	 */
 	detached() {
 		super.detached();
-		this.eventHandler_.removeAllListeners();
+		this.eventHandler_.dispose();
```

Comment:
This change seems wrong.

[`EventHandler` does not implement `dispose()`](https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-events/src/EventHandler.js#L48-L54).

I imagine this will throw an error if you manually test it, which you should try to do before taking it out of draft mode.

Reply:
Good catch @wincent, I'm getting `TypeError: this.eventHandles_[i].removeListener is not a function` when testing this manually with `.removeAllListeners()` alongside the new `delegate`.

That leads me to a conclusion that `.removeAllListeners()` needs to support `.dispose()` as well as `.removeListener()`. Not sure how exactly to implement that, perhaps something like this:
```js
removeAllListeners() {
	for (let i = 0; i < this.eventHandles_.length; i++) {
		const handle = this.eventHandles_[i];

		if (handle.dispose()) {
			handle.dispose();
		}
		else {
			handle.removeListener();
		}
	}

	this.eventHandles_ = [];
}
```
Another idea is to add a `.removeListener()` support to the implementation of the `delegate` utility.

Reply:
Neither of those sound like ~good~ acceptable options:

1. We don't want to touch legacy code in the `metal-events` package.
2. We don't want to clutter up non-legacy code in the `delegate()` implementation to interop with legacy code.

I'd say just drop usage of `EventHandler` from this part of the code base. Search for `eventHandler_` in the file and you'll see it is used for exactly two things. I'd just remove it.

Untested/typed-in-browser:

```js
    attached() {
        this.selectUserButtonHandle_ = dom.on(
            this.selectUserButton,
            'click',
            this._handleSelectUserButtonClicked.bind(this)
        );

        this.removeUserButtonHandle_ = delegate(
            this.container,
            'click',
            this.removeUserLinkSelector,
            this._handleRemoveUserButtonClicked.bind(this)
        );
    }

    detached() {
        super.detached();
        this.selectUserButtonHandle_.removeListener();
        this.removeUserButtonHandle_.dispose();
    }
```